rust_src/admin/pre-commit

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+##
+## To install simply cd to the top level and run:
+## $ ln -s rust_src/admin/pre-commit .git/hooks/pre-commit
+##
+
+STASH_NAME="pre-commit-$(date +%s)"
+
+finish() {
+    [ "$(git stash list --format='%s' | cut -d' ' -f3)" == "$STASH_NAME" ] && git stash pop -q
+}
+
+trap finish EXIT
+
+set -eu
+
+git stash save -q --keep-index $STASH_NAME
+
+check_char='\xE2\x9C\x93'
+cross_char='\xE2\x9D\x8C'
+green='\033[0;32m'
```

Comment:
this could be $(tput setaf 2), so that it's not specific to any particular terminal.

Reply:
Or we could drop it. Either way.